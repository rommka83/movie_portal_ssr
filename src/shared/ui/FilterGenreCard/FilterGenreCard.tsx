import styles from './filtergenrecard.module.css';
import classNames from 'classnames';
interface IFilterGenreCard {
  caption: string;
  genre: string;
  className?: string;
  iconClassName?: string;
  containerClassName?: string;
}
export function FilterGenreCard({
  caption,
  genre,
  className,
  iconClassName,
  containerClassName,
}: IFilterGenreCard) {
  return (
    <div className={classNames(styles.container, containerClassName)}>
      <div className={classNames(styles.containerInner, className)}>
        <div className={classNames(styles.icon, `icon-genre_${genre}_24__0`, iconClassName)} />
        <span className={styles.caption}>{caption}</span>
      </div>
    </div>
  );
}
