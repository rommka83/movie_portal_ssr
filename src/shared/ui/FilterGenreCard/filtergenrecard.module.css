.container {
	display: inline-block;
	width: 25%;
	min-width: 80px;
	padding: 0 6px;
	cursor: pointer;
}

@media (max-width: 500px) {
	.container {
		width: 33%;
	}
}

@media (max-width: 408px) {
	.container {
		width: 50%;
	}
}

.containerInner {
	display: flex;
	flex-direction: column;
	align-items: center;
	gap: 8px;
	border-radius: 6px;
	padding: 16px 8px;
	height: 88px;
	background-color: var(--dark-gray);
	transition: duration 0.4s;
}

.containerInner:active {
	background-color: var(--gray-524);
}

.icon {
	position: relative;
	font-size: 32px;
	line-height: 1;
	color: var(--pink);
}

.icon::after {
	position: absolute;
	top: 0;
	right: 0;
}

.caption {
	transition-duration: 0.4s;
	color: var(--white);
	font-size: 13px;
	line-height: 16px;
	font-weight: 500;
	white-space: nowrap;
}
