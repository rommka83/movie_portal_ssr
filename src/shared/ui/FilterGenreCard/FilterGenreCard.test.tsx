import renderer from 'react-test-renderer';
import { FilterGenreCard } from './FilterGenreCard';

describe('проверка FilterGenreCard', () => {
  test('отрисовка FilterGenreCard', () => {
    const component = renderer.create(<FilterGenreCard caption={''} genre={''} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
