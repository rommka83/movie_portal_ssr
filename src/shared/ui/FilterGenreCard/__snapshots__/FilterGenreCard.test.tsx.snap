// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка FilterGenreCard отрисовка FilterGenreCard 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="containerInner"
  >
    <div
      className="icon icon-genre__24__0"
    />
    <span
      className="caption"
    />
  </div>
</div>
`;
